lib/maze/option/validator.rb

```diff
@@ -74,6 +74,40 @@ def validate_bs(options, errors)
         errors << "--#{Option::ACCESS_KEY} must be specified" if options[Option::ACCESS_KEY].nil?
       end
 
+      # Validates Sauce Labs options
+      def validate_sl(options, errors)
+        # SL local binary
+        sl_local = Maze::Helper.expand_path options[Option::SL_LOCAL]
+        errors << "Sauce Connect binary '#{sl_local}' not found" unless File.exist? sl_local
+
+        # Device
+        browser = options[Option::BROWSER]
+        device = options[Option::DEVICE]
+        os = options[Option::OS]
+        os_version = options[Option::OS_VERSION]
+        if browser.nil? && device.nil? && os.nil? && os_version.nil?
+          errors << "A device or browser option must be specified"
+        elsif browser
+          errors << 'Browsers not yet implemented on Sauce Labs'
+        else
+          # App
+          app = options[Option::APP]
+          if app.nil?
+            errors << "--#{Option::APP} must be provided when running on a device"
+          else
+            uuid_regex = Regexp.new '^[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}$'
```

Comment:
`^` and `$` match the start & end of a line rather than the entire string. This should probably be `\A` and `\z`, which match the start & end of the string — https://rubyapi.org/3.0/o/regexp#class-Regexp-label-Anchors

```suggestion
            uuid_regex = /\A[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}\z/
```